src/popup/router/pages/Retip.vue

```diff
@@ -31,71 +28,118 @@
 
 <script>
 import { mapGetters, mapState } from 'vuex';
-import BigNumber from 'bignumber.js';
-import tipping from 'aepp-raendom/src/utils/tippingContractUtil';
+import tipping from 'tipping-contract/util/tippingContractUtil';
 import { MAGNITUDE, calculateFee, TX_TYPES } from '../../utils/constants';
+import { convertToken } from '../../utils/helper';
 import deeplinkApi from '../../../mixins/deeplinkApi';
 import AmountSend from '../components/AmountSend';
 import UrlStatus from '../components/UrlStatus';
 import Button from '../components/Button';
+import BalanceInfo from '../components/BalanceInfo';
 
 export default {
   mixins: [deeplinkApi],
-  components: { AmountSend, UrlStatus, Button },
+  components: { AmountSend, UrlStatus, Button, BalanceInfo },
   data: () => ({
     tip: {},
     amount: '',
     loading: false,
   }),
   computed: {
     ...mapGetters(['tippingSupported']),
+    ...mapState('fungibleTokens', ['selectedToken']),
     ...mapState({
       tippingV1: 'tippingV1',
+      tippingV2: 'tippingV2',
       urlStatus(state, getters) {
         return getters['tipUrl/status'](this.tip.url);
       },
       isMinTipAmountError(state, { minTipAmount }) {
         return this.amount < minTipAmount;
       },
-      isAmountValid({ sdk, balance }, { account, activeNetwork }) {
-        if (!sdk) return false;
+      isAmountValid({ sdk, balance }, { account }) {
+        if (
+          !sdk ||
+          !this.tippingContract ||
+          (this.selectedToken && this.$route.query.id.includes('_v1'))
+        ) {
+          return false;
+        }
         const fee = calculateFee(TX_TYPES.contractCall, {
           ...sdk.Ae.defaults,
-          contractId: activeNetwork.tipContractV1,
+          contractId: this.tippingContract.deployInfo.address,
           callerId: account.publicKey,
         });
+        if (this.selectedToken) {
+          return (
+            balance - fee >= 0 &&
+            this.selectedToken.convertedBalance > this.amount &&
```

Comment:
```suggestion
            this.selectedToken.convertedBalance >= this.amount &&
```
why I shouldn't be able to send all my tokens?